app/src/main/webapp/app/error/error.ctrl.js

```diff
@@ -206,7 +207,36 @@ esbMessageAdminApp.controller('ErrorCtrl', [
     // End Date picker stuff
 
     $scope.resubmitMessage = function() {
-      alert("Not implemented yet");
+      var message = $scope.message;
+
+      if (!message.allowsResubmit) {
+        messageCenterService.add('warning', 'Message can not be resubmitted', {
+          timeout: 5000
+        });
+
+      } else {
+        var promise = EsbMessageService.resubmitMessage(message);
+
+        promise.then(
```

Comment:
Putting the promise in a local variable is fine, but not very conventional, most of the time I think you would see people just chain `then` the `resubmitMessage(message)`